### PR TITLE
DB/Core: Heroes of old Blizzlike and Fix

### PR DESCRIPTION
**Pull request by wizzymore**
##### CHANGES PROPOSED:

-  Make Heroes of Old blizzlike
proof: https://www.youtube.com/watch?v=p52Jpb1Xu28
In this video -> https://www.youtube.com/watch?v=1dSz4a-g3yE it can be seen when he accepts the quest that the blizzlike uses phases for this quests.

###### ISSUES ADDRESSED:

Closes 

##### TESTS PERFORMED:

- [x] Windows
- [x] Linux

##### HOW TO TEST THE CHANGES:

.go creature 35237
.quest add 2681
.quest complete 2681 and reward it
Take the quest "Hero of old" now, speak with the fallen hero of the horde and complete the quest.( The NPC should be spawned across the road) 
If the player gets disconected or something is messing up his phase, he can click " Hero, i have a urgent business with Thund Splithoof" and the correct phase is set.

##### KNOWN ISSUES AND TODO LIST:

- [ ]
- [ ] 

##### Target branch(es):

Master

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1553731839786525000.sql [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+DELETE FROM `conditions` WHERE  `SourceTypeOrReferenceId`=15 AND `SourceGroup`=840 AND `SourceEntry`=2 AND `SourceId`=0 AND `ElseGroup`=1 AND `ConditionTypeOrReference`=29 AND `ConditionTarget`=0 AND `ConditionValue1`=7750 AND `ConditionValue2`=40 AND `ConditionValue3`=0;
+
+-- Creature Corporal Thund Splithoof 7750 SAI
+SET @ENTRY := 7750;
+UPDATE `creature_template` SET `AIName`="SmartAI" WHERE `entry`= @ENTRY;
+DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY;
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(@ENTRY, 0, 0, 0, 20, 0, 100, 0, 2701, 0, 0, 0, 44, 1, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, "When player rewards quest 2701 - Action invoker: Set phase id to 1 // ");
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1553731839786525000.sql]: Please use single ' for strings, not ".

File: data/sql/updates/pending_db_world/rev_1553731839786525000.sql [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+(@ENTRY, 0, 1, 0, 62, 0, 100, 1, 881, 0, 0, 0, 26, 2801, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, "On gossip action 0 from menu 881 selected - Action invoker: Call event happened from quest 2801 for the whole group // Fallen Hero of the Horde - On Gossip Option 1 Selected - Quest Credit 'A Tale of Sorrow'"),
+(@ENTRY, 0, 2, 0, 61, 0, 100, 0, 0, 0, 0, 0, 72, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, "On link - Action invoker: Close gossip // Fallen Hero of the Horde - On Gossip Option 0 Selected - Close Gossip"),
+(@ENTRY, 0, 3, 0, 61, 0, 100, 0, 0, 0, 0, 0, 72, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, "On link - Action invoker: Close gossip // Fallen Hero of the Horde - On Gossip Option 1 Selected - Close Gossip"),
+(@ENTRY, 0, 4, 0, 19, 0, 100, 0, 2702, 0, 0, 0, 44, 3, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, "When player accepts quest 2702 - Action invoker: Set phase id to 3 // "),
+(@ENTRY, 0, 5, 0, 62, 0, 100, 0, 840, 2, 0, 0, 44, 3, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, "On gossip action 2 from menu 840 selected - Action invoker: Set phase id to 3 // ");
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1553731839786525000.sql]: Same on these lines.

File: data/sql/updates/pending_db_world/rev_1553731839786525000.sql [hunk 3]
```diff
@@ -0,0 +1,30 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1553731839786525000');
+
+DELETE FROM `creature` WHERE `guid`=3110330;
+INSERT INTO `creature` (`guid`, `id`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `modelid`, `equipment_id`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `spawndist`, `currentwaypoint`, `curhealth`, `curmana`, `MovementType`, `npcflag`, `unit_flags`, `dynamicflags`, `ScriptName`, `VerifiedBuild`) VALUES 
+(3110330, 7750, 0, 0, 0, 1, 2, 0, 1, -10630.2, -2988.19, 28.8757, 4.93301, 300, 0, 0, 3399, 0, 0, 0, 0, 0, '', 0);
+
+DELETE FROM `creature` WHERE `guid`=35237;
+INSERT INTO `creature` (`guid`, `id`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `modelid`, `equipment_id`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `spawndist`, `currentwaypoint`, `curhealth`, `curmana`, `MovementType`, `npcflag`, `unit_flags`, `dynamicflags`, `ScriptName`, `VerifiedBuild`) VALUES (35237, 7572, 0, 0, 0, 1, 1, 0, 1, -10632.3, -3009.37, 29.2653, 6.19592, 300, 0, 0, 4121, 0, 0, 0, 0, 0, '', 0);
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1553731839786525000.sql]: Those lines can be combined:

http://www.azerothcore.org/wiki/Dealing-with-SQL-files#write-compact-code

**Comment by poszer** [comment]
@comix1988 Test with this changes please

**Comment by comix1988** [comment]
@poszer @wizzymore 
I just tested this fix and it works perfectly, both for alliance and horde player.
Question: Why does the mission require to open the chest and is still completed automatically? The chest is there but is not spawned.

spectral

gt

**Comment by comix1988** [comment]
wait....after implementing this fix I can't complete the "Fall From Grace" (ID 2784) quest anymore

**Comment by poszer** [comment]
> wait....after implementing this fix I can't complete the "Fall From Grace" (ID 2784) quest anymore.
> If I restart the server I can complete it with a pg but with the second pg I can not complete it anymore. Could it be that it is set that it can be completed only once for the whole server?

Try with new character, and do quests once, do not repeat them on character please. @comix1988

**Comment by comix1988** [comment]
I have tried with various pg even different accounts
@poszer

**Comment by wizzymore** [comment]
@comix1988 
```sql
(@ENTRY, 0, 0, 0, 62, 0, 100, 1, 842, 0, 0, 0, 26, 2784, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'On gossip action 0 from menu 842 selected - Action invoker: Call event happened from quest 2784 for the whole group // Fallen Hero of the Horde - On Gossip Option 0 Selected - Quest Credit \'Fall From Grace\''),
(@ENTRY, 0, 1, 0, 62, 0, 100, 1, 881, 0, 0, 0, 26, 2801, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'On gossip action 0 from menu 881 selected - Action invoker: Call event happened from quest 2801 for the whole group // Fallen Hero of the Horde - On Gossip Option 1 Selected - Quest Credit \'A Tale of Sorrow\''),
```
This completes the quest.
If you do
```sql
SELECT * FROM gossip_menu_option WHERE menuid = 842 OR menuid = 881;
```
you get 2 menu options
842 with option 0
881 with option 0

https://imgur.com/a/fOo3xfB

Check this link out before giving reviews on a PR -> http://www.azerothcore.org/wiki/How-to-test-DB-only-changes

As for `Question: Why does the mission require to open the chest and is still completed automatically? The chest is there but is not spawned.`. Because that is not the quest i am fixing today. That quest autocompletes.

**Comment by comix1988** [comment]
I confirm that the mission is working properly.
I will open a new issue for the mission "fall from grace".

good job @wizzymore I'm sorry for the misunderstanding.

**Pull request merged**